### PR TITLE
Add types for rox-node

### PR DESCRIPTION
**Pull request by g-guirado**
Submitting the PR internally before submitting it to the official repo.
Differences with rox-browser:
- doc links
- showOverrides doesn't exist
- removed freeze/unfreeze

### CODE REVIEW

File: types/rox-node/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  // The name of the Flag
+  readonly name: string;
+
+  // Default value of the Flag
+  readonly defaultValue: boolean;
+
+  // Returns true when the flag is enabled
+  isEnabled(): boolean;
+
+  getNameDetails(): RoxNameDetails;
```
Comment (ekeren) [inline types/rox-node/index.d.ts]: This should be a private function.
1. please do not document it
2. please rename it (add _)

File: types/rox-node/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  // The name of the Variant
+  readonly name: string;
+
+  // Default value of the Variant
+  readonly defaultValue: BasicType<T>;
+
+  // Returns the current value of the Variant, accounting for value overrides
+  getValue(): BasicType<T>;
+
+  getNameDetails(): RoxNameDetails;
```
Comment (ekeren) [inline types/rox-node/index.d.ts]: This should be a private function.

please do not document it
please rename it (add _)

File: types/rox-node/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+}
+
+interface RoxNameDetails {
```
Comment (ekeren) [inline types/rox-node/index.d.ts]: redundant

**Review by ekeren — CHANGES_REQUESTED** [review]

File: types/rox-node/index.d.ts (hunk 1 above)
Comment (g-guirado) [reply types/rox-node/index.d.ts]: removed

**Comment by g-guirado** [comment]
Will open a PR for this on DefinitelyTypes once this one is through: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38452

**Pull request closed without merging**